core/formula/src/main/java/org/csstudio/apputil/formula/areadetector/ImageOfFunction.java

```diff
@@ -0,0 +1,132 @@
+import org.epics.util.array.ArrayFloat;
+import org.epics.util.array.ArrayInteger;
+import org.epics.util.array.ArrayLong;
+import org.epics.util.array.ArrayShort;
+import org.epics.util.array.ListNumber;
+import org.epics.vtype.Alarm;
+import org.epics.vtype.Display;
+import org.epics.vtype.Time;
+import org.epics.vtype.VImage;
+import org.epics.vtype.VImageDataType;
+import org.epics.vtype.VImageType;
+import org.epics.vtype.VNumberArray;
+import org.epics.vtype.VType;
+import org.phoebus.core.vtypes.VTypeHelper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+/** A formula function for creating an VImage
+ *  @author Kunal Shroff
+ */
+public class ImageOfFunction implements FormulaFunction
+{
+    @Override
+    public String getCategory()
+    {
+        return "areaDetector";
+    }
+
+    @Override
+    public String getName()
+    {
+        return "imageOf";
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Create an VImage from array data";
+    }
+
+    @Override
+    public List<String> getArguments()
+    {
+        return List.of("imageData", "height", "width",
+                "imageDataType (pvInt, pvUInt, pvDouble)", "vImageType (TYPE_MONO, TYPE_BAYER, TYPE_RGB1)",
+                "xoffset", "yoffset", "xreversed", "yreversed");
+    }
+
+    @Override
+    public VType compute(final VType... args) throws Exception
+    {
+        if (args.length != 9)
+            throw new Exception("Function " + getName() +
+                    " takes 9 arguments but received " + args.length);
+
+        VType value = args[0];
+
+        int height = Double.valueOf(VTypeHelper.toDouble(args[1])).intValue();
+        int width = Double.valueOf(VTypeHelper.toDouble(args[2])).intValue();
+
+
+        final VImageDataType imageDataType =
+                VImageDataType.getVImageDataType(VTypeHelper.toString(args[3]));
+        if(imageDataType == null) {
+            throw new Exception("Failed to parse the image data type from the provided argument " + args[3]);
+        }
+
+        // Get data and data type
+        VImageType vImageType = getVImageType(VTypeHelper.toString(args[4]));
+        if(imageDataType == null) {
+            throw new Exception("Failed to parse the image type from  " + args[4]);
+        }
+        final VNumberArray data;
+        if (VTypeHelper.isNumericArray(value)) {
+            data = (VNumberArray) value;
+        } else {
+            throw new Exception("Failed to parse the image data array.");
+        }
+
+        int xoffset = Double.valueOf(VTypeHelper.toDouble(args[5])).intValue();
+        int yoffset = Double.valueOf(VTypeHelper.toDouble(args[6])).intValue();
+
+        boolean xreversed = Boolean.parseBoolean(VTypeHelper.toString(args[7]));
+        boolean yreversed = Boolean.parseBoolean(VTypeHelper.toString(args[8]));
+
+
+        Alarm alarm = Alarm.none();
+        Time time = Time.now();
```

Comment:
Maybe there's a way to copy the time stamp from the image data array? That time stamp might be important.

Reply:
same with alarm info.  Admittedly, for the image widget it doesn't much matter, we plot the data and ignore the time stamp. This is more about future usage of `=imageOf(....`

Reply:
done